packages/shylu/shylu_dd/frosch/src/InterfaceSets/FROSch_InterfaceEntity_def.hpp

```diff
@@ -329,7 +329,7 @@ namespace FROSch {
                             for (UN k=0; k<NodeVector_.size(); k++) {
                                 // Coordinates of the nodes of the entity
                                 SCVecPtr E(dimension);
-                                SC distance = 0.0;
+                                SC distance = Teuchos::ScalarTraits<SC>::zero();
                                 // Compute quadratic distance
                                 for (UN l=0; l<dimension; l++) {
                                     distance += (nodeList->getData(l)[this->getLocalNodeID(k)]-CN[l]) * (nodeList->getData(l)[this->getLocalNodeID(k)]-CN[l]);
```

Comment:
This will be very slow.  Don't use `getData` in an inner loop if you can avoid it.  Consider `withLocalAccess` (see above).

Reply:
Here is a test for `withLocalAccess` with Vectors: https://github.com/trilinos/Trilinos/blob/ddd82db72f34ffa280bf9f1a54b3946d38f9a8fe/packages/tpetra/core/test/MultiVector/WithLocalAccess.cpp#L429

Reply:
You might also want to look at `transform`: https://github.com/trilinos/Trilinos/blob/master/packages/tpetra/core/test/MultiVector/Transform.cpp

Reply:
@mhoemmen Does something like this exists for Xpetra as well?

Reply:
@searhein No, Xpetra doesn't have such a thing.  I'm not sure if Xpetra is allowed to expose Kokkos::View even, since it may still need to build with Kokkos disabled.